core/src/main/java/com/distkv/core/concepts/DistkvConcepts.java

```diff
@@ -1,35 +1,38 @@
 package com.distkv.core.concepts;
 
-import com.distkv.common.exception.DistkvKeyDuplicatedException;
 import com.distkv.common.exception.KeyNotFoundException;
 import com.distkv.common.utils.Status;
 import com.distkv.core.DistkvMapInterface;
 
 public abstract class DistkvConcepts<T> {
 
   // The Reference of the key value map.
-  protected DistkvMapInterface<String, Object> distkvKeyValueMap;
+  protected DistkvMapInterface<String, DistkvValue> distkvKeyValueMap;
 
-  protected DistkvConcepts(DistkvMapInterface<String, Object> distkvKeyValueMap) {
+  protected DistkvConcepts(DistkvMapInterface<String, DistkvValue> distkvKeyValueMap) {
     this.distkvKeyValueMap = distkvKeyValueMap;
   }
 
-  public void put(String key, T value) {
-    if (distkvKeyValueMap.containsKey(key)) {
-      throw new DistkvKeyDuplicatedException(key);
-    }
-    distkvKeyValueMap.put(key, value);
-  }
+  public abstract void put(String key, T value);
```

Comment:
Why make this as abstract?

Reply:
If abstract is not used in this place, then subclass implementation. This is not very easy to achieve. Because it is not clear what value object is passed in. If the DistkvValue object is passed uniformly, all api puts are a bit problematic. The first is that you must pass a DistkvValue with a type, but this type is not needed, because the semantics of the api has determined the type.